stellar_sdk/server.py

```diff
@@ -39,17 +39,20 @@ class Server(BaseServer):
         (ex. ``"https://horizon-testnet.stellar.org"`` for test network,
         ``"https://horizon-testnet.stellar.org"`` for public network)
     :param client: Http client used to send the request
+    :param custom_request_headers: Any headers that should be added when making HTTP requests to the
+    client.
     """
 
     def __init__(
         self,
         horizon_url: str = "https://horizon-testnet.stellar.org/",
         client: BaseSyncClient = None,
+        custom_request_headers: Optional[Dict[str, str]] = None,
```

Comment:
please check https://github.com/StellarCN/py-stellar-base/pull/599#discussion_r895083961

Reply:
@overcat I certainly understand that perspective, but here's a counterpoint: Right now in my own code, I've basically had to replicate about 50 lines of code from this file to make my own `Session`, just to be able to add the one additional header in my requests. It is an unnecessary burden on users in such a situation and leads to a poor experience. You'll also find that more consumers of this library might need such a feature, as there are more services out there in the market that let you query Horizon via APIs that require authentication.

Basically, the cost of adding this feature is nothing and the downside is placing an unnecessary burden on users. It seems like an easy win to me.

Reply:
Hi @RohitK89, I'm not sure why we need to add 50 lines of code, is the following code not enough to do the job?
```python
from stellar_sdk import Server
from stellar_sdk.client.requests_client import RequestsClient

client = RequestsClient(custom_headers={"key": "value"})
server = Server("https://horizon-testnet.stellar.org", client)
```

I think `custom_request_headers` is part of `client`, so I still don't want to add it here. If we add it here, should we also add other parameters(`pool_size`, `num_retries`, `request_timeout` etc) here?

Reply:
Hi @RohitK89, I'm not sure why we need to add 50 lines of code, is the following code not enough to do the job?
```python
from stellar_sdk import Server
from stellar_sdk.client.requests_client import RequestsClient

client = RequestsClient(custom_headers={"key": "value"})
server = Server("https://horizon-testnet.stellar.org", client)
```

I think custom_request_headers is part of client, so I still don't want to add it here. If we add it here, should we also add other parameters(pool_size, num_retries, request_timeout etc) here?

Reply:
oh, sorry, I misunderstood. I see what you mean - yes, I think it's not unreasonable to add the param only in the client and let users instantiate their own clients and pass to the server. I'll update.